spotlight-client/src/charts/ResponsiveTooltipController.tsx

```diff
@@ -70,29 +79,32 @@ const ResponsiveTooltipController: React.FC<ResponsiveTooltipControllerProps> =
   setHighlighted,
   customHoverBehavior,
 }) => {
-  const infoPanelDispatch = useInfoPanelDispatch();
-  const infoPanelState = useInfoPanelState();
+  const { uiStore } = useDataStore();
   const enableInfoPanel = useBreakpoint(false, ["mobile-", true]);
   const [clickAnnotations, setClickAnnotations] = useState<AnnotationType[]>();
 
-  useEffect(() => {
-    // when state is cleared, make sure any relevant chart props are cleared also
-    if (isEmpty(infoPanelState)) {
-      if (setHighlighted) setHighlighted();
-      setClickAnnotations(undefined);
-    }
-  }, [infoPanelState, setHighlighted]);
+  useEffect(
+    () =>
+      autorun(() => {
+        // when state is cleared, make sure any relevant chart props are cleared also
+        if (isEmpty(uiStore.infoPanelData)) {
+          if (setHighlighted) setHighlighted();
+          setClickAnnotations(undefined);
+        }
+      }),
+    [setHighlighted, uiStore]
+  );
```

Comment:
@daschi I wonder if this would've helped with our tooltip and tick memory issue on the charts when receiving a new set of data. Maybe we just needed to clear them manually...

I think our solution is fine, just thought you might want to see this.